From: Benjamin Barenblat <bbaren@google.com>
Subject: Set package configuration options
Forwarded: not-needed

Configure Abseil for Debian.

  - Set the SONAME appropriately.

  - To minimize the possibility of future ABI breakage, treat absl::any,
    absl::optional, absl::string_view, and absl::variant as their own types
    (rather than aliases for the std:: versions), and compile everything in an
    inline namespace.

  - Enable upstream's hardened build mode.

Index: abseil/CMake/AbseilHelpers.cmake
===================================================================
--- abseil.orig/CMake/AbseilHelpers.cmake
+++ abseil/CMake/AbseilHelpers.cmake
@@ -314,6 +314,7 @@ Cflags: -I\${includedir}${PC_CFLAGS}\n")
       set_target_properties(${_NAME} PROPERTIES
         OUTPUT_NAME "absl_${_NAME}"
         SOVERSION "${ABSL_SOVERSION}"
+        VERSION "${ABSL_SOVERSION}.0.0"
       )
     endif()
   else()
Index: abseil/absl/base/options.h
===================================================================
--- abseil.orig/absl/base/options.h
+++ abseil/absl/base/options.h
@@ -97,7 +97,7 @@
 // absl::string_view is a typedef of std::string_view, use the feature macro
 // ABSL_USES_STD_STRING_VIEW.
 
-#define ABSL_OPTION_USE_STD_STRING_VIEW 2
+#define ABSL_OPTION_USE_STD_STRING_VIEW 0
 
 // ABSL_OPTION_USE_STD_ORDERING
 //
@@ -179,6 +179,6 @@
 // checks enabled by this option may abort the program in a different way and
 // log additional information when `NDEBUG` is not defined.
 
-#define ABSL_OPTION_HARDENED 0
+#define ABSL_OPTION_HARDENED 1
 
 #endif  // ABSL_BASE_OPTIONS_H_
